ui/pages/notifications-settings/notifications-settings-per-account.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useEffect, useState, useCallback } from 'react';
+import { useSelector } from 'react-redux';
+import { useMetamaskNotificationsContext } from '../../contexts/metamask-notifications/metamask-notifications';
+import {
+  useSwitchAccountNotifications,
+  useSwitchAccountNotificationsChange,
+  type UseSwitchAccountNotificationsData,
+} from '../../hooks/metamask-notifications/useSwitchNotifications';
+import {
+  NotificationsSettingsBox,
+  NotificationsSettingsAccount,
+} from '../../components/multichain';
+import { getIsUpdatingMetamaskNotificationsAccount } from '../../selectors/metamask-notifications/metamask-notifications';
+
+type NotificationsSettingsPerAccountProps = {
+  address: string;
+  name: string;
+  disabled: boolean;
+  loading: boolean;
+};
+
+export const NotificationsSettingsPerAccount = ({
+  address,
+  name,
+  disabled,
+  loading,
+}: NotificationsSettingsPerAccountProps) => {
+  const { listNotifications } = useMetamaskNotificationsContext();
+
+  // Hooks
+  const {
+    onChange: onChangeAccountNotifications,
+    error: errorAccountNotificationsChange,
+  } = useSwitchAccountNotificationsChange();
+  const { switchAccountNotifications, error: errorSwitchAccountNotifications } =
+    useSwitchAccountNotifications();
+
+  const [data, setData] = useState<
+    UseSwitchAccountNotificationsData | undefined
+  >(undefined);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  // Selectors
+  const isUpdatingMetamaskNotificationsAccount = useSelector(
+    getIsUpdatingMetamaskNotificationsAccount,
+  );
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const fetchedData = await switchAccountNotifications([address]);
+      setData(fetchedData || {});
+    };
+    fetchData();
+  }, [address, switchAccountNotifications]);
+
+  useEffect(() => {
+    setIsLoading(loading);
+  }, [loading]);
+
+  useEffect(() => {
+    const updateData = async () => {
+      if (isUpdatingMetamaskNotificationsAccount.includes(address)) {
+        const fetchedData = await switchAccountNotifications([address]);
+        setData(fetchedData || {});
+      }
+    };
+    updateData();
+  }, [isUpdatingMetamaskNotificationsAccount]);
```

Comment:
Confirm that these effects do work? I'm not so convinced we need them all (but maybe we do).

If the notification settings page is starting to become a pain, lets just deliver MVP/minimum.

Reply:
If it works it works, we can tidy this up later.